packages/@aws-cdk/aws-stepfunctions-tasks/lib/batch/run-batch-job.ts

```diff
@@ -170,6 +172,8 @@ export interface RunBatchJobProps {
 
 /**
  * A Step Functions Task to run AWS Batch
+ *
+ * @deprecated use `BatchSubmitJob`
```

Comment:
Can we reduce the duplication on this implementaion? Can `RunBatchJob` call `BatchSubmitJob`?

Reply:
similar [comment]((https://github.com/aws/aws-cdk/pull/8178#discussion_r434282770)) to the one I made for Step Functions. I don't think it's as simple as a straight up swap.

I do agree, there's probably room for some refactoring. I did not try to do it all in this PR though.